insights/parsers/lspci.py

```diff
@@ -1,57 +1,135 @@
-    ... 00:02.0 VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Controller (rev 09)
-    ... 03:00.0 Network controller: Intel Corporation Centrino Advanced-N 6205 [Taylor Peak] (rev 34)
-    ... 0d:00.0 System peripheral: Ricoh Co Ltd PCIe SDXC/MMC Host Controller (rev 07)
-    ... '''.strip()
-    >>> from insights.tests import context_wrap
-    >>> shared = {LsPci: LsPci(context_wrap(lspci_content))}
-    >>> pci_info = shared[LsPci]
-    >>> pci_info.get("Intel Corporation")[0]['raw_message']
-    '00:00.0 Host bridge: Intel Corporation 2nd Generation Core Processor Family DRAM Controller (rev 09)', '00:02.0 VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Controller (rev 09)', '03:00.0 Network controller: Intel Corporation Centrino Advanced-N 6205 [Taylor Peak] (rev 34)'
-    >>> len(pci_info.get("Network controller"))
-    1
-    >>> "Centrino Advanced-N 6205" in pci_info
-    True
-    >>> "0d:00.0" in pci_info
-    True
+Lists ALL the PCI Devices With Details
+======================================
+
+Parsers included in this module are:
+
+LsPci - command ``/sbin/lspci``
+-------------------------------
+
+LsPciDriver - command ``/sbin/lspci -k``
+----------------------------------------
+
 """
-from .. import LogFileOutput, parser, CommandParser
+
+from .. import LogFileOutput, parser, CommandParser, get_active_lines
 from insights.specs import Specs
 
 
+@parser(Specs.lspci_kernel)
+class LsPciDriver(CommandParser):
+    """
+    This module provides methods to access driver information of PCI devices from
+    the ``/sbin/lspci -k`` command.
+
+    Typical output of the ``lspci -k`` command is::
+
+        00:00.0 Host bridge: Intel Corporation 5500 I/O Hub to ESI Port (rev 13)
+                Subsystem: Cisco Systems Inc Device 0101
+        00:01.0 PCI bridge: Intel Corporation 5520/5500/X58 I/O Hub PCI Express Root Port 1 (rev 13)
+                Kernel driver in use: pcieport
+                Kernel modules: shpchp
+        00:02.0 PCI bridge: Intel Corporation 5520/5500/X58 I/O Hub PCI Express Root Port 2 (rev 13)
+                Kernel driver in use: pcieport
+                Kernel modules: shpchp
+        06:00.0 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)
+                Subsystem: Cisco Systems Inc Device 004a
+                Kernel driver in use: ixgbe
+                Kernel modules: ixgbe
+        06:00.1 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)
+                Subsystem: Cisco Systems Inc Device 004a
+                Kernel driver in use: ixgbe
+                Kernel modules: ixgbe
+
+    Example:
+
+        >>> lspcidriv = shared[LsPciDriver]
+        >>> pci_devs = lspcidriv.pci_dev_details.keys()
```

Comment:
I mean you could add an interface to list all the installed devices, instead of calling the `keys()` of the dict. e.g.

```python
def list_pci_devices(self):
  return lspcidriv.pci_dev_details.keys()

def __contain__(self, dev):
  return dev in lspcidriv.pci_dev_details.keys()

```